web-components/src/components/robotoff-modal/robotoff-modal.ts

```diff
@@ -0,0 +1,169 @@
+import { LitElement, html, nothing } from "lit"
+import { customElement, property, state } from "lit/decorators.js"
+import { EventState, EventType, RobotoffContributionType } from "../../constants"
+import { BasicStateEventDetail } from "../../types"
+import "../shared/modal"
+import "../robotoff-ingredients/robotoff-ingredients"
+import "../robotoff-nutrients/robotoff-nutrients"
+import "../robotoff-question/robotoff-question"
+import { localized, msg } from "@lit/localize"
+
+/**
+ * The `robotoff-modal` component is a web component that displays a modal for contributing to product information.
+ * It handles different types of contributions (ingredients, nutrients, and questions) and manages the modal's state.
+ *
+ * @fires success - Dispatched when a contribution is successfully made.
+ * @fires close - Dispatched when the modal is closed.
+ *
+ * @example
+ * ```html
+ * <robotoff-modal product-code="123456789" robotoff-contribution-type="ingredients"></robotoff-modal>
+ * ```
+ */
+
+@customElement("robotoff-modal")
+@localized()
+export class RobotoffModal extends LitElement {
+  /**
+   * The type of contribution being made.
+   */
+  @property({ type: String, attribute: "robotoff-contribution-type" })
+  robotoffContributionType?: RobotoffContributionType
+
+  /**
+   * The product code for which the contribution is being made.
+   */
+  @property({ type: String, attribute: "product-code" })
+  productCode: string = ""
+
+  /**
+   * Indicates whether the modal is in a loading state.
+   */
+  @state()
+  isLoading = false
+
+  /**
+   * Indicates whether the success message should be shown.
+   */
+  @state()
+  showSuccessMessage = false
+
+  /**
+   * Returns whether the modal is open based on the `robotoffContributionType`.
+   */
+  get isOpen() {
+    return Boolean(this.robotoffContributionType)
+  }
+
+  /**
+   * Dispatches a success event when a contribution is successfully made.
+   * @param {RobotoffContributionType} type - The type of contribution.
+   */
+  sendSuccessEvent(type: RobotoffContributionType) {
+    this.dispatchEvent(
+      new CustomEvent(EventType.SUCCESS, {
+        bubbles: true,
+        composed: true,
+        detail: {
+          type: type,
+        },
+      })
+    )
+  }
+
+  /**
+   * Handles the success event by showing a success message and dispatching a success event.
+   */
+  onSuccessEvent() {
+    const robotoffContributionType = this.robotoffContributionType!
+    this.showSuccessMessage = true
+    setTimeout(() => {
+      this.showSuccessMessage = false
+      this.sendSuccessEvent(robotoffContributionType)
+    }, 1000)
+  }
+
+  /**
+   * Handles state changes from child components.
+   * @param {CustomEvent<BasicStateEventDetail>} event - The state change event.
+   */
+  onStateChange(event: CustomEvent<BasicStateEventDetail>) {
+    switch (event.detail.state) {
+      case EventState.ANNOTATED:
+        this.onSuccessEvent()
+        break
```

Comment:
Is "skip" bubling an ANNOTATED event ?

Reply:
Ok it seems ANNOTATED is sent when every insight have been processed. I thikn we should document it a bit more. (although in constant.ts it's well documented).

Or maybe we should rename to ALL_ANNOTATED ?